rest-api/src/main/java/io/quarkus/qe/exceptions/CatalogExceptionMapper.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.qe.exceptions;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import static javax.ws.rs.core.Response.Status;
+import static javax.ws.rs.core.Response.status;
+
+@Provider
+public class CatalogExceptionMapper implements ExceptionMapper<Exception> {
+
+    @Override
+    public Response toResponse(Exception e) {
+
+        if (e instanceof RepositoryAlreadyExistsException) {
+            RepositoryAlreadyExistsException exp = (RepositoryAlreadyExistsException) e;
+            return status(Status.CONFLICT).entity(exp.toCatalogError(e.getMessage())).build();
+        }
+
+        if (e instanceof RepositoryNotFoundException) {
+            RepositoryNotFoundException exp = (RepositoryNotFoundException) e;
+            return status(Status.NOT_FOUND).entity(exp.toCatalogError(e.getMessage())).build();
```

Comment:
This seems weird to me: "exp.toCatalogError(e.getMessage())"

The method toCatalogError belongs to this class (It's not logic of the CatalogException exception) and then just do:

```
toCatalogError(exp);
```